src/main/update/launcher-update.ts

```diff
@@ -116,34 +107,40 @@ export async function update(update: Update, listeners: IUpdateOptions) {
   console.log("peerVersionExtra (bytes):", buffer);
   const extra = decode(buffer) as BencodexDict;
   console.log("peerVersionExtra (decoded):", JSON.stringify(extra)); // Stringifies the JSON for extra clarity in the log
-  const macOSBinaryUrl = extra.get("macOSBinaryUrl") as string;
-  const macOSPlayerBinaryUrl = extra.get("macOSPlayerBinaryUrl") as string;
-  const windowsPlayerBinaryUrl = extra.get("WindowsPlayerBinaryUrl") as string;
-  const windowsBinaryUrl = extra.get("WindowsBinaryUrl") as string;
-
-  console.log("macOSBinaryUrl: ", macOSBinaryUrl);
-  console.log("WindowsBinaryUrl: ", windowsBinaryUrl);
-  console.log("macOSPlayerBinaryUrl: ", macOSPlayerBinaryUrl);
-  console.log("WindowsPlayerBinaryUrl: ", windowsPlayerBinaryUrl);
-
-  const downloadUrl =
-    process.platform === "win32"
-      ? windowsBinaryUrl
-      : process.platform === "darwin"
-      ? macOSBinaryUrl
-      : null;
-  const playerDownloadUrl =
-    process.platform === "win32"
-      ? windowsPlayerBinaryUrl
-      : process.platform === "darwin"
-      ? macOSPlayerBinaryUrl
-      : null;
-
-  if (downloadUrl == null) {
-    console.log(`Stop update process. Not support ${process.platform}.`);
-    return;
+
+  // FIXME: project version number hard coding: 1.
+  const network = getConfig("Network", "9c-main");
+
+  let netenv;
+  if (network === "9c-main") {
+    netenv = "main";
+  } else {
+    netenv = network;
   }
 
+  const launcherDownloadUrl = getDownloadUrl(
+    netenv,
+    peerVersionNumber,
+    "launcher",
+    1,
+    process.platform
+  );
+  const playerDownloadUrl = getDownloadUrl(
+    netenv,
+    peerVersionNumber,
+    "player",
+    1,
```

Comment:
It seems like figuring out the project version is out of scope for this PR. So take this as a reminder.

Reply:
https://github.com/planetarium/9c-launcher/issues/1744
I'll update the body later